src/Crypto/WebAuthn/Cose/PublicKeyWithSignAlg.hs

```diff
@@ -53,8 +62,18 @@ data PublicKeyWithSignAlg = PublicKeyWithSignAlgInternal
 -- field.
 type CosePublicKey = PublicKeyWithSignAlg
 
+-- | Binary representation of the COSE_Messages structure defined in the [spec](https://datatracker.ietf.org/doc/html/rfc8152#section-2).
+newtype CoseMessage = CoseMessage {unCoseMessage :: BS.ByteString}
+  deriving newtype (Eq, Show)
+  deriving (ToJSON) via PrettyHexByteString
+
+-- | Binary representation of the COSE_Signature structure in the [spec](https://datatracker.ietf.org/doc/html/rfc8152#section-4.1).
+newtype CoseSignature = CoseSignature {unCoseSignature :: BS.ByteString}
+  deriving newtype (Eq, Show)
+  deriving (ToJSON) via PrettyHexByteString
+
```

Comment:
Taking a closer look at the specs again, I don't think this is right. The encoding of these depends on the signature algorithm used, see [here](https://www.w3.org/TR/webauthn-2/#sctn-signature-attestation-types). It particular it doesn't seem to be using the `COSE_Messages` and `COSE_Signature` structures at all.

```suggestion
-- | A wrapper for the bytes of a message that should be verified.
-- This is used for both assertion and assertion.
newtype Message = Message {unMessage :: BS.ByteString}
  deriving newtype (Eq, Show)
  deriving (ToJSON) via PrettyHexByteString

-- | [(spec)](https://www.w3.org/TR/webauthn-2/#sctn-signature-attestation-types)
-- A wrapper for the bytes of a signature that can be used to verify a 'Message'.
-- The encoding is specific to webauthn and depends on the 'CoseSignAlg' used.
newtype Signature = Signature {unSignature :: BS.ByteString}
  deriving newtype (Eq, Show)
  deriving (ToJSON) via PrettyHexByteString

```

Reply:
From what I understand, webauthn mainly relies on the COSE spec _just_ for the public key and signing algorithm (see `COSE_Key` [here](https://www.w3.org/TR/webauthn-2/#credentialpublickey)), that's really all that's implemented by the `Crypto.WebAuthn.Cose` modules. Other than that, the COSE spec isn't needed, the `COSE_Sign` and co. are never used for webauthn.